backend/src/routes/auth.js

```diff
@@ -148,29 +152,32 @@ these checks then return the user the token they can use to make requests
 to other endpoints.
 */
 async function handle2FactorAuthentication(req, res) {
-  const { email, sessionId, code } = req.body;
-  if (!sessionId || !code) {
-    return res.status(400).send({ error: 'Invalid request' });
+  const { sessionId, code } = req.body;
+  const { email } = req.locals.user;
+  if (!sessionId || !code || !email) {
+    return res.status(400).send({ error: "Invalid request" });
   }
   try {
-    const firebaseRes = await db.collection('2_fac')
-      .doc(email)
-      .get();
+    const firebaseRes = await db.collection("2_fac").doc(email).get();
     const twoFacEntry = firebaseRes.data();
     if (sessionId === twoFacEntry.sessionId && code === twoFacEntry.code) {
       return res.status(200).send({ token: twoFacEntry.token });
     }
-    return res.status(401).send({ error: 'Incorrect code' });
+    return res.status(401).send({ error: "Incorrect code" });
   } catch (_) {
-    return res.status(401).send({ error: 'No session found' });
+    return res.status(401).send({ error: "No session found" });
   }
 }
 
 function getAuthRoutes() {
   const router = express.Router();
-  router.post('/register', handleRegister);
-  router.post('/login', handleLogin);
-  router.post('/2fac', handle2FactorAuthentication);
+  router.post("/register", handleRegister);
+  router.get("/init2facSession", handleLogin);
```

Comment:
lol this camelcase w numbers in the endpoint looks ugly af

Reply:
lmao sadge. maybe we can rename all the routes in some standardized way in a later commit